### PR TITLE
Feature/frontend routing and left side menu

### PR DESCRIPTION
**Pull request by Khrapal-Max**
Created module Auth and Main. Auth is responsible for login verification. Main is responsible for navigation and access to the main components.
Route tree

     - / app
     - / login
     - / menu-nav
         - / menu-home
         - / search
         - / playlist
         - / songs

Added a guard. The layout of the menu has been formed.

### CODE REVIEW

File: frontend/angular.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
               "src/assets"
             ],
             "styles": [
+              "./node_modules/@angular/material/prebuilt-themes/pink-bluegrey.css",
```
Comment (NikitaKrasnovSW) [inline frontend/angular.json]: What is the reason of using Material? We have Semantic UI

File: frontend/package.json [hunk 2]
```diff
@@ -11,10 +11,12 @@
   "private": true,
   "dependencies": {
     "@angular/animations": "~12.1.0-",
+    "@angular/cdk": "^12.1.3",
     "@angular/common": "~12.1.0-",
     "@angular/compiler": "~12.1.0-",
     "@angular/core": "~12.1.0-",
     "@angular/forms": "~12.1.0-",
+    "@angular/material": "^12.1.3",
```
Comment (NikitaKrasnovSW) [inline frontend/package.json:20]: Probably don't need it

File: frontend/src/app/common/material-components.module.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import {MatSliderModule} from '@angular/material/slider';
+import {MatSlideToggleModule} from '@angular/material/slide-toggle';
+import {MatSnackBarModule} from '@angular/material/snack-bar';
+import {MatSortModule} from '@angular/material/sort';
+import {MatTableModule} from '@angular/material/table';
+import {MatTabsModule} from '@angular/material/tabs';
+import {MatToolbarModule} from '@angular/material/toolbar';
+import {MatTooltipModule} from '@angular/material/tooltip';
+import {MatTreeModule} from '@angular/material/tree';
+import {OverlayModule} from '@angular/cdk/overlay';
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/common/material-components.module.ts]: Delete Material components

File: frontend/src/app/guard/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  canActivateChild(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+    return this.canActivate(route, state);
+  }
+
+  checkLogin(url: string): boolean {
+    if (this.authService.isLoggedIn) { return true; }
+
+    this.authService.redirectUrl = url;
+
+    const sessionId = 123456789;
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/guard/auth.guard.ts]: Delete hard coded value. If this is a workaround, then write it correctly without extra code

File: frontend/src/app/moduls/auth/landingpage/landingpage.component.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  selector: 'app-landingpage',
+  templateUrl: './landingpage.component.html',
+  styleUrls: ['./landingpage.component.sass']
+})
+export class LandingpageComponent implements OnInit {
+  message: string;
+
+  constructor(public authService: AuthService, public router: Router) {
+    this.message = this.getMessage();
+  }
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/moduls/auth/landingpage/landingpage.component.ts:16]: Add new line

File: frontend/src/app/moduls/main/main-menu/main-menu.component.html [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            <strong>Create Playlist</strong>
+          </button>
+        </div>
+      </div>
+    </mat-sidenav>
+    <mat-sidenav-content>
+      <router-outlet></router-outlet>
+    </mat-sidenav-content>
+  </mat-sidenav-container>
+</div>
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/moduls/main/main-menu/main-menu.component.html]: Replace markdown with Semantic

File: frontend/src/app/services/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  providedIn: 'root',
+})
+export class AuthService {
+  isLoggedIn = false;
+
+  redirectUrl: string | null = null;
+
+  login(): Observable<boolean> {
+    return of(true).pipe(
+      delay(1000),
```
Comment (NikitaKrasnovSW) [inline frontend/src/app/services/auth.service.ts:19]: If this is not a real code, leave TODO comment

**Review by NikitaKrasnovSW — CHANGES_REQUESTED** [review]

File: frontend/angular.json (hunk 1 above)
Comment (Khrapal-Max) [reply frontend/angular.json]: Done

File: frontend/package.json (hunk 2 above)
Comment (Khrapal-Max) [reply frontend/package.json:20]: Deleted " @ angular / material " : " ^ 12.1.3 "

File: frontend/src/app/common/material-components.module.ts (hunk 3 above)
Comment (Khrapal-Max) [reply frontend/src/app/common/material-components.module.ts]: Done

File: frontend/src/app/moduls/auth/landingpage/landingpage.component.ts (hunk 5 above)
Comment (Khrapal-Max) [reply frontend/src/app/moduls/auth/landingpage/landingpage.component.ts:16]: Done

File: frontend/src/app/services/auth.service.ts (hunk 7 above)
Comment (Khrapal-Max) [reply frontend/src/app/services/auth.service.ts:19]: Done.

Added in service
// TODO: Temporary service to simulate authorization and test routing. Remove after installing the correct authorization system.

File: frontend/src/app/guard/auth.guard.ts (hunk 4 above)
Comment (Khrapal-Max) [reply frontend/src/app/guard/auth.guard.ts]: Done.
Added TODO for method checkLogin
// TODO: Temporary body in metod.
  // After adding the token based authorization. 
  // Here, in the verification condition, replace the request for the authorization service - token verification.

File: frontend/src/app/moduls/main/main-menu/main-menu.component.html (hunk 6 above)
Comment (Khrapal-Max) [reply frontend/src/app/moduls/main/main-menu/main-menu.component.html]: Done

**Review by Khrapal-Max — COMMENTED** [review]
Edits made

**Review by NikitaKrasnovSW — APPROVED** [review]

**Pull request merged**